crates/astria-core/src/sequencerblock/v1/block/mod.rs

```diff
@@ -620,38 +631,49 @@ enum SequencerBlockHeaderErrorKind {
 /// Exists to provide convenient access to fields of a [`SequencerBlock`].
 #[derive(Clone, Debug, PartialEq)]
 pub struct SequencerBlockParts {
-    pub block_hash: [u8; 32],
+    pub block_hash: [u8; SHA256_DIGEST_LENGTH],
     pub header: SequencerBlockHeader,
     pub rollup_transactions: IndexMap<RollupId, RollupTransactions>,
     pub rollup_transactions_proof: merkle::Proof,
     pub rollup_ids_proof: merkle::Proof,
-    pub extended_commit_info: Option<Bytes>,
-    pub extended_commit_info_proof: Option<merkle::Proof>,
+    pub upgrade_change_hashes_with_proof: Option<UpgradeChangeHashesWithProof>,
+    pub extended_commit_info_with_proof: Option<ExtendedCommitInfoWithProof>,
 }
 
 pub struct SequencerBlockBuilder {
-    pub block_hash: [u8; 32],
+    pub block_hash: [u8; SHA256_DIGEST_LENGTH],
     pub chain_id: tendermint::chain::Id,
     pub height: tendermint::block::Height,
     pub time: Time,
     pub proposer_address: account::Id,
-    pub data: Vec<Bytes>,
+    pub parsed_data_items: ParsedDataItems,
     pub deposits: HashMap<RollupId, Vec<Deposit>>,
-    pub with_extended_commit_info: bool,
 }
 
 impl SequencerBlockBuilder {
     /// Attempts to build a [`SequencerBlock`] from the provided data.
     ///
     /// # Errors
     ///
-    /// - if the first transaction in the data is not the 32-byte rollup transactions root.
-    /// - if the second transaction in the data is not the 32-byte rollup IDs root.
-    /// - if `with_extended_commit_info` is set to `true` and the third transaction in the data is
-    ///   not the extended commit info.
-    /// - if any transaction in the data cannot be decoded.
-    /// - if the rollup transactions root does not match the reconstructed root.
-    /// - if the rollup IDs root does not match the reconstructed root.
+    /// Returns an error if `uses_data_item_enum` is false and `with_extended_commit_info` is true,
```

Comment:
i don't see `uses_data_item_enum` in this file anywhere other than this comment?

Reply:
This comment needs a full overhaul - I'll fix it up thanks.

Reply:
Done in [4a045d3](https://github.com/astriaorg/astria/pull/1917/commits/4a045d36c177676c3af883aed37da3cce61f30b6).